samples/azure-quantum/CONTRIBUTING.md

```diff
@@ -0,0 +1,40 @@
+# Process for creating notebook samples for the Azure Quantum portal experience
```

Comment:
In general, this file has a lot of Markdownlint issues that should be resolved — I've made some concrete comments with suggestions for some of them, but it'd be worth making sure we don't have any warnings before merging in. Thanks!